File: Silver/All/SharedUI/SharedUI.Shared/MainWindowController.swift
open __partial class MainWindowController {

	private func setup() {
		//
		//  Do any shared initilaization, here
		//
	}

	//
	// Add Shared code here
	//

	@Notify public var valueA: String!
	@Notify public var valueB: String!
	@Notify public private(set) var result: String!

	@IBAction
	public func calculateResult(_ sender: id!) {
		if (length(valueA) == 0) | (length(valueB) == 0) {
			result = "(value required)"
		} else {
			if let a = Convert.TryToDoubleInvariant(valueA), let b = Convert.TryToDoubleInvariant(valueB) {
				result = Convert.ToString(a + b)
			} else {
				result = valueA + valueB
			}
		}
	}
}

File: Oxygene/All/SharedUI/SharedUI.Shared/MainWindowController.pas
namespace SharedUI.Shared;

type
  MainWindowController = public partial class
  public

    //
    // Add Shared code here
    //

    [Notify] property valueA: String;
    [Notify] property valueB: String;
    [Notify] property &result: String read private write;

    [IBAction]
    method calculateResult(aSender: id);
    begin
      if (length(valueA) = 0) or (length(valueB) = 0) then begin
        &result := '(value required)';
      end
      else begin
        var a := Convert.TryToDoubleInvariant(valueA);
        var b := Convert.TryToDoubleInvariant(valueB);
        if assigned(a) and assigned(b) then begin
          &result := Convert.ToString(a + b);
        end
        else begin
          &result := valueA + valueB;
        end;
      end;
    end;

  private

    method setup;
    begin
      //
      //  Do any shared initilaization, here
      //
      //
    end;

  end;

end.